ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -42,6 +48,8 @@ tests = testGroup "Mempool"
   , testProperty "getTxs == getTxsAfter zeroIdx"        prop_Mempool_getTxs_getTxsAfter
   , testProperty "addedTxs == getTxs"                   prop_Mempool_addTxs_getTxs
   , testProperty "Invalid transactions are never added" prop_Mempool_InvalidTxsNeverAdded
+  , testProperty "Added valid transactions are traced"  prop_Mempool_TraceValidTxs
+  , testProperty "Rejected invalid txs are traced"      prop_Mempool_TraceRejectedTxs
```

Comment:
Make an issue to make sure we have a test that would have caught the fact that in this PR we do not `syncState` before getting the mempool snapshot. This test may not strictly speaking be a mempool test, since it's about the use of the mempool in the consensus. Nevertheless, it's an important test.